spinalcordtoolbox/math.py

```diff
@@ -113,14 +106,61 @@ def dilate(data, size, shape, dim=None):
     """
     if isinstance(data, Image):
         im_out = data.copy()
-        im_out.data = dilate(data.data, size, shape, dim)
+        im_out.data = dilate(data.data, size, shape, dim, islabel)
         return im_out
     else:
         footprint = _get_footprint(shape, size, dim)
-        if data.dtype in ['uint8', 'uint16']:
-            return rank.maximum(data, footprint=footprint)
+        if islabel:
+            return _dilate_point_labels(data, footprint=footprint)
         else:
-            return dilation(data, footprint=footprint, out=None)
+            if data.dtype in ['uint8', 'uint16']:
+                return rank.maximum(data, footprint=footprint)
+            else:
+                return dilation(data, footprint=footprint)
+
+
+def _dilate_point_labels(data, footprint):
+    """
+    A more efficient dilation algorithm when we know the image is mostly zero (i.e. point label image)
+    """
+    if len(data.shape) != len(footprint.shape):
+        raise ValueError(f"incompatible shapes: {data.shape=} {footprint.shape=}")
+    if footprint.size == 0:
+        raise ValueError(f"cannot handle empty footprint: {footprint.shape=}")
+
+    data_shape = np.array(data.shape)
+    fp_shape = np.array(footprint.shape)
+
+    # For odd-length dimensions, the center of the footprint is unique.
+    # For even-length dimensions, we round down.
+    fp_center = (fp_shape - 1) // 2
```

Comment:
It seems silly to say `x - 1 + 1` instead of just `x`, but `x // 2` really looks like rounding _down_, which would be confusing.
```suggestion
    # For even-length dimensions, we round up.
    fp_center = (fp_shape - 1 + 1) // 2
```